chain/chain/src/resharding.rs

```diff
@@ -66,23 +91,58 @@ fn apply_delayed_receipts<'a>(
     Ok(new_state_roots)
 }
 
+// function to set up flat storage status to Ready after a resharding event
+// TODO(shreyan) : Consolidate this with setting up flat storage during state sync logic
+fn set_flat_storage_state(
+    store: Store,
+    flat_storage_manager: &FlatStorageManager,
+    shard_uid: ShardUId,
+    block_info: BlockInfo,
+) -> Result<(), Error> {
+    assert!(flat_storage_manager.get_flat_storage_for_shard(shard_uid).is_none());
```

Comment:
what assumption are we making here with assert?

Reply:
Just realized this same check exists in `create_flat_storage_for_shard` as well, so I'll just remove this.